### ISSUE TITLE
[feature] super(); this.prop = value; not shrinked as super().prop = value

### ISSUE THREAD

**Issue by WebReflection** [terser/terser#1322]
*TL;DR* terser perfectly understands `super().prop = value` syntax but it's not collapsing it when `super(); this.prop = value;` is used in the code. As `super()` always returns the current context, I'd expect if the instantly next operation is a property assignment, *terser* should know how to shrink that: see the example at the bottom.

**Bug report or Feature request?**

This is rather a feature request but it's possible terser has a bug if the current outcome is not intended.

**Version (complete output of `terser -V` or specific git commit)**

5.16.1

**Complete CLI command or `minify()` options used**

`npx terser ./input.js -m -c`

**`terser` input**

```js
class A {}
class B {
  constructor() {
    super();
    this.shrinked = false;
  }
}
```

**`terser` output or error**

```js
class A{}class B{constructor(){super(),this.shrinked=!1}}
```

**Expected result**

```js
class A{}class B{constructor(){super().shrinked=!1}}
```

**Comment by WebReflection** [comment]
P.S. if anyone is wondering "*why don't you just write `super().prop` then?*" the reason is that *TS* is not that smart so I need to write code like:

```js
  constructor(fn, value) {
    super(value);

    /**
     * Disposes the computed effect to stop receiving updates.
     * @type {() => void}
     */
    this.dispose = effect(() => {
      super.value = fn(this.peek());
    });
  }
```

because if I write the comment before `super(value).dispose = ...` the information is completely lost or messed up.

I'd love to have that shrinked once minified though, hope this makes sense.

**Comment by fabiosantoscode** [comment]
Smells like usignal code :) 

Terser doesn't have the concept of "this expression always returns `this`", and this is a pretty tiny compression opportunity so it's unlikely this would be worked on (bang vs buck considerations with maintenance cost and compression).

So I'm closing but a workaround is to leverage a variable. In this case Terser doesn't have to know what the variable returns, and the compression you expect still happens:

```
var this_ = super()
this_.shrinked = false
```

I think you may be able to use the `@method` jsdoc directive to specify `dispose` is a method of that class.
